core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java

```diff
@@ -0,0 +1,90 @@
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.earlyreturn;
+
+import io.temporal.client.*;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+
+public class EarlyReturnClient {
+  private static final String TASK_QUEUE = "EarlyReturnTaskQueue";
+  private static final String WORKFLOW_ID_PREFIX = "early-return-workflow-";
+
+  public static void main(String[] args) {
+    WorkflowClient client = setupWorkflowClient();
+    runWorkflowWithUpdateWithStart(client);
+  }
+
+  // Set up the WorkflowClient
+  public static WorkflowClient setupWorkflowClient() {
+    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
+    return WorkflowClient.newInstance(service);
+  }
+
+  // Run workflow using 'updateWithStart'
+  private static void runWorkflowWithUpdateWithStart(WorkflowClient client) {
+    TransactionRequest txRequest =
+        new TransactionRequest(
+            "Bob", "Alice",
+            -1000); // Change this amount to a negative number to have initTransaction fail
+
+    WorkflowOptions options = buildWorkflowOptions();
+    TransactionWorkflow workflow = client.newWorkflowStub(TransactionWorkflow.class, options);
+
+    System.out.println("Starting workflow with UpdateWithStart");
+
+    UpdateWithStartWorkflowOperation<TxResult> updateOp =
+        UpdateWithStartWorkflowOperation.newBuilder(workflow::returnInitResult)
+            .setWaitForStage(WorkflowUpdateStage.COMPLETED) // Wait for update to complete
+            .build();
+
+    TxResult updateResult = null;
+    try {
+      WorkflowUpdateHandle<TxResult> updateHandle =
+          WorkflowClient.updateWithStart(workflow::processTransaction, txRequest, updateOp);
+
+      updateResult = updateHandle.getResultAsync().get();
+
+      System.out.println(
+          "Workflow initialized with result: "
+              + updateResult.getStatus()
+              + " (transactionId: "
+              + updateResult.getTransactionId()
+              + ")");
+
+      TxResult result = WorkflowStub.fromTyped(workflow).getResult(TxResult.class);
+      System.out.println(
+          "Workflow completed with result: "
+              + result.getStatus()
+              + " (transactionId: "
+              + result.getTransactionId()
+              + ")");
+    } catch (Exception e) {
+      System.err.println("Transaction initialization failed: " + e.getMessage());
+    }
+  }
+
+  // Build WorkflowOptions with task queue and unique ID
+  private static WorkflowOptions buildWorkflowOptions() {
+    return WorkflowOptions.newBuilder()
+        .setTaskQueue(TASK_QUEUE)
+        .setWorkflowId(WORKFLOW_ID_PREFIX + System.currentTimeMillis())
```

Comment:
In the Go example, I'm generated the transaction ID on the client and use it in the Workflow ID. That way, the user would receive an error when they start a workflow for the same transaction again. I don't think it's strictly necessary to mimic that here, but I thought I'd mention it.

Reply:
Gotcha. I intentionally changed it so the early-return would contain meaningful data usable for the rest of the transaction (an early-return / initialization step that returns null isn't very illustrative of real-world value imo)

I plan on following Quinn's suggestion here of still having the ID minted as part of an activity, but doing it in a separate one as part of the initialization step https://github.com/temporalio/samples-java/pull/689#discussion_r1817000501 -- Let me know if there's anything you'd do differently